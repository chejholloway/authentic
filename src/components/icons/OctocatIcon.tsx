import React from "react";
import octocat from "../../assets/images/github-octocat-svgrepo-com.svg";
/**
 * Renders the Octocat icon as an image.
 *
 * This component displays the Octocat logo as an image. You can provide an optional
 * CSS class name to style the icon as needed.
 *
 * @component
 * @example
 * ```tsx
 * <OctocatIcon className="text-gray-600" />
 * ```
 * @defaultprops
 * @param {object} props - The component props.
 * @param {string} [props.className] - An optional CSS class name to apply custom styling.
 */
interface OctocatIconProps {
  className?: string;
}

const OctocatIcon: React.FC<OctocatIconProps> = ({ className }) => {
  return (
    <img
      className={`h-10 object-cover ${className}`}
      src={octocat}
      alt="Github Octocat"
    />
  );
};

export default OctocatIcon;
