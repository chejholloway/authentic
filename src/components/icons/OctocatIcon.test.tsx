import { render } from '@testing-library/react';
import React from 'react';
import octocat from "../../assets/images/github-octocat-svgrepo-com.svg";
import OctocatIcon from "./OctocatIcon";

jest.mock("../../assets/images/github-octocat-svgrepo-com.svg");

describe('<OctocatIcon>', () => {
  it('should render component', () => {
    const { container } = render(<OctocatIcon />);

    expect(container).toMatchSnapshot();
  });

  it('should render component with props', () => {
    const { container } = render(<OctocatIcon className="dummy-class" />);

    expect(container).toMatchSnapshot();
  });
});
