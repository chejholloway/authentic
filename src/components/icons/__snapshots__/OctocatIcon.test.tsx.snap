// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OctocatIcon> should render component 1`] = `
<div>
  <img
    alt="Github Octocat"
    class="h-10 object-cover undefined"
    src="github-octocat-svgrepo-com.svg"
  />
</div>
`;

exports[`<OctocatIcon> should render component with props 1`] = `
<div>
  <img
    alt="Github Octocat"
    class="h-10 object-cover dummy-class"
    src="github-octocat-svgrepo-com.svg"
  />
</div>
`;
